api/server/router/image/image_routes.go

```diff
@@ -472,12 +472,12 @@ func (ir *imageRouter) getImagesSearch(ctx context.Context, w http.ResponseWrite
 
 func (ir *imageRouter) postImagesPrune(ctx context.Context, w http.ResponseWriter, r *http.Request, vars map[string]string) error {
 	if err := httputils.ParseForm(r); err != nil {
-		return err
+		return errdefs.InvalidParameter(err)
 	}
 
 	pruneFilters, err := filters.FromJSON(r.Form.Get("filters"))
 	if err != nil {
-		return err
+		return errdefs.InvalidParameter(err)
```

Comment:
filters.FromJSON already returns typed errors, so this should probably not be changed https://github.com/moby/moby/blob/980d8e5d54d8df31ba0e75e1d9dd474d9d91308a/api/types/filters/parse.go#L85